### PR TITLE
STORM-3440: client: fix all checkstyle warnings

### PR DESCRIPTION
**Pull request by krichter722**

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/trident/Stream.java [hunk 1]
```diff
@@ -349,15 +355,15 @@ public TridentState partitionPersist(StateFactory stateFactory, Fields inputFiel
 
     public TridentState partitionPersist(StateSpec stateSpec, Fields inputFields, StateUpdater updater, Fields functionFields) {
         projectionValidation(inputFields);
-        String id = _topology.getUniqueStateId();
-        ProcessorNode n = new ProcessorNode(_topology.getUniqueStreamId(),
-                                            _name,
+        String id = topology.getUniqueStateId();
+        ProcessorNode n = new ProcessorNode(topology.getUniqueStreamId(),
+                name,
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/trident/Stream.java]: Shouldn't the other params move into this level as well?

File: storm-client/src/jvm/org/apache/storm/trident/TridentTopology.java [hunk 2]
```diff
@@ -224,6 +226,7 @@ private static SpoutNode getDRPCSpoutNode(Collection<Node> g) {
     }
 
     private static void checkValidJoins(Collection<Node> g) {
+        @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
         boolean hasDRPCSpout = false;
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/trident/TridentTopology.java]: Can rename

File: storm-client/src/jvm/org/apache/storm/trident/TridentTopology.java [hunk 3]
```diff
@@ -177,6 +177,7 @@ private TridentTopology(DefaultDirectedGraph<Node, IndexedEdge> graph, Map<Strin
         return ret;
     }
 
+    @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
     private static void completeDRPC(DefaultDirectedGraph<Node, IndexedEdge> graph, Map<String, List<Node>> colocate, UniqueIdGen gen) {
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/trident/TridentTopology.java]: private, can rename

File: storm-client/src/jvm/org/apache/storm/trident/TridentTopology.java [hunk 4]
```diff
@@ -211,6 +212,7 @@ private static Node getLastAddedNode(Collection<Node> g) {
     }
 
     //returns null if it's not a drpc group
+    @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
     private static SpoutNode getDRPCSpoutNode(Collection<Node> g) {
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/trident/TridentTopology.java]: private, can rename

File: storm-client/src/jvm/org/apache/storm/trident/topology/TridentBoltExecutor.java [hunk 5]
```diff
@@ -177,43 +177,43 @@ private void checkFinish(TrackedBatch tracked, Tuple tuple, TupleType type) {
...
         //get transaction id
         //if it already exists and attempt id is greater than the attempt there
 
 
-        TrackedBatch tracked = (TrackedBatch) _batches.get(id.getId());
-        //        if(_batches.size() > 10 && _context.getThisTaskIndex() == 0) {
-        //            System.out.println("Received in " + _context.getThisComponentId() + " " + _context.getThisTaskIndex()
-        //                    + " (" + _batches.size() + ")" +
+        TrackedBatch tracked = (TrackedBatch) batches.get(id.getId());
+        //        if(batches.size() > 10 && context.getThisTaskIndex() == 0) {
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/trident/topology/TridentBoltExecutor.java]: I think we should delete the commented out code here

File: storm-client/src/jvm/org/apache/storm/utils/Utils.java [hunk 6]
```diff
@@ -1234,6 +1227,7 @@ public static void validateTopologyBlobStoreMap(Map<String, Object> topoConf, Bl
      */
     public static String threadDump() {
         final StringBuilder dump = new StringBuilder();
+        @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
         final java.lang.management.ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/utils/Utils.java]: Can rename

File: storm-server/src/main/java/org/apache/storm/DaemonConfig.java [hunk 7]
```diff
@@ -1015,62 +1015,62 @@
...
      */
-    @isImplementationOfClass(implementsClass = IHttpCredentialsPlugin.class)
+    @ConfigValidationAnnotations.IsImplementationOfClass(implementsClass = IHttpCredentialsPlugin.class)
     public static final String UI_HTTP_CREDS_PLUGIN = "ui.http.creds.plugin";
 
     /**
      * Class name of the HTTP credentials plugin for DRPC.
      */
-    @isImplementationOfClass(implementsClass = IHttpCredentialsPlugin.class)
+    @ConfigValidationAnnotations.IsImplementationOfClass(implementsClass = IHttpCredentialsPlugin.class)
```
Comment (srdo) [inline storm-server/src/main/java/org/apache/storm/DaemonConfig.java]: I think some of these qualifications aren't necessary

File: storm-client/src/jvm/org/apache/storm/utils/Utils.java [hunk 8]
```diff
@@ -1321,7 +1315,7 @@ public static double nullToZero(Double v) {
      * @param b something else
      * @return a or b the first one that is not null
      */
-    public static <V> V OR(V a, V b) {
+    public static <V> V or(V a, V b) {
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/utils/Utils.java]: Public API, we should suppress instead

File: storm-client/src/jvm/org/apache/storm/Config.java [hunk 9]
```diff
@@ -1771,38 +1774,47 @@ private static List getRegisteredDecorators(Map<String, Object> conf) {
         return ret;
     }
 
+    @SuppressWarnings("checkstyle:OverloadMethodsDeclarationOrder")
```
[the remark this answers is not in the document]
Comment (krichter722) [reply storm-client/src/jvm/org/apache/storm/Config.java]: Do you prefer static and non-static methods overloading each other next to each other? Or should static remain separated from non-static?

[the remark this answers is not in the document]
Comment (srdo) [reply storm-client/src/jvm/org/apache/storm/Config.java]: I don't have an opinion on this. Do what makes sense to you. If you'd rather keep the suppressions, it's fine IMO.

File: storm-client/src/jvm/org/apache/storm/state/BaseStateIterator.java [hunk 10]
```diff
@@ -23,15 +23,15 @@
 /**
  * Base implementation of iterator over {@link KeyValueState}. Encoded/Decoded types of key and value are all generic.
  */
-public abstract class BaseStateIterator<K, V, KENCODED, VENCODED> implements Iterator<Map.Entry<K, V>> {
+public abstract class BaseStateIterator<K, V, KENCODEDT, VENCODEDT> implements Iterator<Map.Entry<K, V>> {
```
[the remark this answers is not in the document]
Comment (krichter722) [reply storm-client/src/jvm/org/apache/storm/state/BaseStateIterator.java]: Exactly.

File: storm-client/src/jvm/org/apache/storm/bolt/JoinBolt.java [hunk 11]
```diff
@@ -75,11 +73,12 @@ public JoinBolt withOutputStream(String streamName) {
...
     /**
-     * Performs inner Join with the newStream. SQL    :   from priorStream inner join newStream on newStream.field = priorStream.field1 same
-     * as:   new WindowedQueryBolt(priorStream,field1). join(newStream, field, priorStream);
-     *
-     * Note: priorStream must be previously joined. Valid ex:    new WindowedQueryBolt(s1,k1). join(s2,k2, s1). join(s3,k3, s2); Invalid ex:
-     * new WindowedQueryBolt(s1,k1). join(s3,k3, s2). join(s2,k2, s1);
+     * Performs inner Join with the newStream. SQL:
+     * <code>from priorStream inner join newStream on newStream.field = priorStream.field1 same
+     * as:   new WindowedQueryBolt(priorStream,field1). join(newStream, field, priorStream);</code>
+     * Note: priorStream must be previously joined. Valid ex:
```
[the remark this answers is not in the document]
Comment (krichter722) [reply storm-client/src/jvm/org/apache/storm/bolt/JoinBolt.java]: No idea why Github doesn't display this outdated, see https://github.com/krichter722/storm/blob/checkstyle-client/storm-client/src/jvm/org/apache/storm/bolt/JoinBolt.java for the actual version

File: storm-client/src/jvm/org/apache/storm/trident/Stream.java (hunk 1 above)
Comment (krichter722) [reply storm-client/src/jvm/org/apache/storm/trident/Stream.java]: This is fixed as well, but not displayed outdated.

**Comment by srdo** [comment]
Please put review fixes in a new commit instead of amending the previous commit. I don't want to look at all 12k lines again, and Github's force push diff doesn't seem that good (currently it doesn't seem to show any of the most recent changes you've done).

You can probably find the original commit in your `git reflog`, and get the most recent changes by diffing that commit to the current commit in this PR.

**Comment by srdo** [comment]
Sorry if we've been unclear about this, we used to tell people to squash at the end of reviews, because we used to merge outside Github. Since we can now squash and merge through Github, there's no reason for people to amend commits anymore.

File: docs/STORM-UI-REST-API.md [hunk 12]
```diff
@@ -743,7 +743,7 @@ Sample response:
         "java.library.path": "/usr/local/lib:/opt/local/lib:/usr/lib",
         "topology.executor.send.buffer.size": 1024,
         "storm.local.dir": "storm-local",
-        "storm.messaging.netty.buffer_size": 5242880,
+        "storm.messaging.netty.bufferSize": 5242880,
```
[the remark this answers is not in the document]
Comment (srdo) [reply docs/STORM-UI-REST-API.md]: This is still showing as changed for me

File: storm-client/src/jvm/org/apache/storm/StormSubmitter.java [hunk 13]
```diff
@@ -71,14 +72,14 @@ public static boolean validateZKDigestPayload(String payload) {
         return false;
     }
 
+    @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
```
[the remark this answers is not in the document]
Comment (srdo) [reply storm-client/src/jvm/org/apache/storm/StormSubmitter.java]: Same comment as before, the diff hasn't changed.

File: storm-client/src/jvm/org/apache/storm/security/auth/authorizer/SimpleWhitelistAuthorizer.java [hunk 14]
```diff
@@ -22,12 +22,13 @@
 /**
  * An authorization implementation that simply checks a whitelist of users that are allowed to use the cluster.
  */
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
 public class SimpleWhitelistAuthorizer implements IAuthorizer {
```
[the remark this answers is not in the document]
Comment (srdo) [reply storm-client/src/jvm/org/apache/storm/security/auth/authorizer/SimpleWhitelistAuthorizer.java]: Same comment as before.

**Comment by srdo** [comment]
Thanks, the changes look good. I unresolved the couple of comments I think weren't addressed.

**Comment by srdo** [comment]
Thanks, +1 assuming tests pass.

**Comment by krichter722** [comment]
You're welcome. Thanks for your review.

I assumed that Github is capable of showing changes after amended and force-pushed commits. FYI, I rebased the first commit on master for easy conflict resolution (I'm sure that https://github.com/krichter722/storm/compare/%20ec527b6..6a2955192aac2f31cd6a0d5ec60210fb7dd8dbe7 shows that there're no changes) and pushed further commits on top.

**Pull request merged**